### ISSUE TITLE
Update README.md

### ISSUE THREAD

**Issue by svekars** [pachyderm/pachyderm#3827]
Rewrite, reorganize, and improve the main README.md file.
A good example of README.md:

https://github.com/windmilleng/tilt

**Comment by svekars** [comment]
@JoeyZwicker is there something specific that you want to see in the README? I compared to the linked README by Tilt, and to be honest, it is very similar to ours. Would love some details.

**Comment by JoeyZwicker** [comment]
You're right that its not that different. Mainly, I think I wrote the Readme like a year ago and wanted to make sure it still conformed to how we want to pitch ourselves. I thnk the main section that Tilt does super well that we don't is the contributor guide, but that's likely something we'll need to eng to do.

**Comment by JoeyZwicker** [comment]
feel free to close or deprioritize this unless there are any obvious places you think we can improve (potentially looking at other major OSS projects) and stealing ideas from them

**Comment by svekars** [comment]
I will do a basic editorial cleanup and will see if I have better ideas later.

**Comment by mindthevirt** [comment]
We want to add getting started steps to the readme, similar to hear: https://www.pachyderm.com/getting-started/
